File: Exp3/app/src/main/java/com/example/exp3/MainActivity.java
package com.example.exp3;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private int counter = 0;
    private TextView textCounter;
    private Button btnIncrement, btnReset;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        textCounter = findViewById(R.id.textCounter);
        btnIncrement = findViewById(R.id.btnIncrement);
        btnReset = findViewById(R.id.btnReset);

        // Increment button
        btnIncrement.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                counter++;
                textCounter.setText(String.valueOf(counter));
            }
        });

        // Reset button
        btnReset.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                counter = 0;
                textCounter.setText(String.valueOf(counter));
            }
        });
    }
}

File: app/src/main/java/com/example/counterapp/MainActivity.kt
package com.example.counterapp

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.TextView

class MainActivity : AppCompatActivity() {

    // Declare a variable to hold the current count
    private var count: Int = 0

    // Declare UI elements
    private lateinit var counterDisplay: TextView
    private lateinit var incrementButton: Button
    private lateinit var decrementButton: Button
    private lateinit var resetButton: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // Find and assign UI elements from the layout file
        counterDisplay = findViewById(R.id.counter_display)
        incrementButton = findViewById(R.id.increment_btn)
        decrementButton = findViewById(R.id.decrement_btn)
        resetButton = findViewById(R.id.reset_btn)

        // Set up click listeners for the buttons
        incrementButton.setOnClickListener {
            count++
            updateDisplay()
        }

        decrementButton.setOnClickListener {
            count--
            updateDisplay()
        }

        resetButton.setOnClickListener {
            count = 0
            updateDisplay()
        }

        // Initial update of the display
        updateDisplay()
    }

    /**
     * A private function to update the TextView with the current count value.
     */
    private fun updateDisplay() {
        counterDisplay.text = count.toString()
    }
}
